components/eamxx/docs/common/installation.md

```diff
@@ -1,9 +1,172 @@
+
+Change to your `$RUN_ROOT_DIR` directory and use CMake to configure your build.
+
+If you're building SCREAM on one of our supported platforms, you can tell CMake
+to use the appropriate machine file using the `-C` flag. Machine files are
+located in `$EAMXX_SRC_DIR/components/eamxx/cmake/machine-files`. Take a look
+and see whether your favorite machine has one.
+
+For example, to configure SCREAM on the Quartz machine at LLNL:
+
+```
+cd $RUN_ROOT_DIR
+cmake \
+    -DCMAKE_CXX_COMPILER=$(which mpicxx) \
+    -DCMAKE_BUILD_TYPE=Debug \
+    -C ${EAMXX_SRC_DIR}/components/eamxx/cmake/machine-files/quartz.cmake \
+    ${EAMXX_SRC_DIR}/components/eamxx
+```
+
+If you're building on a machine that doesn't have a ready-made machine file,
+you can try configuring your build by manually passing options to CMake. This
+usually looks something like the following, which configures EAMxx to compile
+CPU code using Kokkos's OpenMP backend:
+```
+cd $RUN_ROOT_DIR
+cmake \
+    -D CMAKE_BUILD_TYPE=Debug \
+    -D CMAKE_C_COMPILER=mpicc \
+    -D CMAKE_CXX_COMPILER=mpicxx \
+    -D CMAKE_Fortran_COMPILER=mpif90 \
+    -D MPIEXEC_EXECUTABLE=`which mpiexec` \
+    -D EKAT_MPI_NP_FLAG:STRING=-n \
+    -D SCREAM_CIME_BUILD=OFF \
+    -D SCREAM_DYNAMICS_DYCORE=HOMME \
+    -D SCREAM_DOUBLE_PRECISION:BOOL=ON \
+    -D SCREAM_INPUT_ROOT:PATH=/path/to/scream-input \
+    -D Kokkos_ENABLE_DEBUG=TRUE \
+    -D Kokkos_ENABLE_AGGRESSIVE_VECTORIZATION=OFF \
+    -D Kokkos_ENABLE_SERIAL=ON \
+    -D Kokkos_ENABLE_OPENMP=ON \
+    -D Kokkos_ENABLE_LIBDL=OFF \
+    -D Kokkos_ENABLE_PROFILING=OFF \
+    -D Kokkos_ENABLE_DEPRECATED_CODE=OFF \
+    -D KOKKOS_ENABLE_ETI:BOOL=OFF \
+    -D NetCDF_C_PATHS=/path/to/netcdf-c-dir \
+    -D NetCDF_Fortran_PATHS=/path/to/netcdf-f90-dir \
+    -D PnetCDF_C_PATHS=/path/to/pnetcdf-dir \
+    -D PnetCDF_Fortran_PATHS=/path/to/pnetcdf-f90-dir \
+    ${EAMXX_SRC_DIR}/components/eamxx
+```
+
+In either case, EAMxx requires MPI-aware compilers. Let's examine these
+options (only some of which are required on any given machine) to make sure we
+know what they do:
+
+* `CMAKE_BUILD_TYPE`: specifies whether you are building EAMxx in a
+  developer-friendly configuration (`Debug`), for a production run (`Release`)
+  or for performance profiling or some other specialized purpose. Typically,
+  you'll set this option to `Debug` or `Release`.
+* `CMAKE_{C,CXX,Fortran}_COMPILER`: the name of the command used to invoke an
+  MPI-enabled C, C++, or Fortran compiler to build EAMxx
+* `MPIEXEC_EXECUTABLE`: the name of the command used to run EAMxx using MPI,
+  typically `mpiexec` or `mpirun`, but possibly different depending on your
+  desired machine
+* `EKAT_MPI_NP_FLAG`: the flag passed to `MPIEXEC_EXECUTABLE` that you use to
+  specify the number of desired MPI processes. This is typically `-n` for
+  `mpiexec` and `-np` for `mpirun`.
+* `SCREAM_CIME_BUILD`: indicates whether EAMxx uses [CIME](https://e3sm.org/resources/tools/other-tools/cime/)
```

Comment:
The user should not set this entry by hand, we handle this internally. We should not even mention it.

Reply:
Good to know!